waspc/cli/src/Wasp/Cli/Command/CreateNewProject/Templates.hs

```diff
@@ -0,0 +1,79 @@
+module Wasp.Cli.Command.CreateNewProject.Templates
+  ( getStarterTemplates,
+    StarterTemplateNames,
+    templatesToList,
+    isValidTemplateName,
+  )
+where
+
+import Control.Exception (try)
+import Data.Aeson (FromJSON, parseJSON, withObject, (.:))
+import qualified Network.HTTP.Simple as HTTP
+
+data StarterTemplateNames = StarterTemplateNames [String]
+
+getStarterTemplates :: IO (Maybe StarterTemplateNames)
+getStarterTemplates = do
+  -- Github returns 403 if we don't specify user-agent.
+  let request = HTTP.addRequestHeader "User-Agent" "wasp-lang/wasp" templatesRepoInfoURL
+  responseOrException <- try $ HTTP.httpJSONEither request
+  return $ case responseOrException of
+    Right response -> extractTemplateNames . HTTP.getResponseBody $ response
+    Left (_ :: HTTP.HttpException) -> Nothing
+  where
+    templatesRepoInfoURL :: HTTP.Request
+    templatesRepoInfoURL = "https://api.github.com/repos/wasp-lang/starters/git/trees/main"
+
+    -- Each folder in the "wasp-lang/starters" repo is a template.
+    extractTemplateNames :: Either HTTP.JSONException RepoInfo -> Maybe StarterTemplateNames
+    extractTemplateNames (Left _) = Nothing
+    extractTemplateNames (Right body) = Just . StarterTemplateNames . map _path . filter isFolder $ _objects body
+
+    isFolder :: RepoObject -> Bool
+    isFolder = (== Folder) . _type
+
+templatesToList :: StarterTemplateNames -> [String]
+templatesToList (StarterTemplateNames templates) = templates
+
+isValidTemplateName :: String -> StarterTemplateNames -> Bool
```

Comment:
Sounds like you are checking if it is correctly formatted to be a template name. But you are actually checking if it is one of the existing templates. Not terrible, but a bit misleading.